tests/units/reporter/test_md_reporter.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""Test anta.reporter.md_reporter.py."""
+
+from __future__ import annotations
+
+import json
+from io import StringIO
+from pathlib import Path
+
+import pytest
+
+from anta.reporter.md_reporter import MDReportBase, MDReportGenerator
+from anta.result_manager import ResultManager
+from anta.result_manager.models import TestResult as FakeTestResult
+
+DATA_DIR: Path = Path(__file__).parent.parent.parent.resolve() / "data"
+
+
+@pytest.mark.parametrize(
+    ("only_failed_tests", "expected_report_name"),
+    [
+        pytest.param(True, "test_md_report_only_failed_tests.md", id="only_failed_tests"),
+        pytest.param(False, "test_md_report_all_tests.md", id="all_tests"),
+    ],
+)
+def test_md_report_generate(tmp_path: Path, expected_report_name: str, *, only_failed_tests: bool) -> None:
+    """Test the MDReportGenerator class."""
+    # Create a temporary Markdown file
```

Comment:
it does not create the file right?

Reply:
No. `tmp_path` is a pytest fixture --> https://docs.pytest.org/en/stable/how-to/tmp_path.html

Reply:
so let's remove this comment

Reply:
Fixed in fb2139a3fd7763022aec4746db3a350aea9404c3